### PR TITLE
[wip] Centcom Departure Lounge

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
add: All late arrivals must be processed through Centcom Departures before reaching the station. This is a great chance to meet and mingle with your coworkers.
/:cl:

image
image
- Adds Centcom Departures as the new late join spawn location, which contains a small maint section, and a bunch of wardrobes and stuff, things you'd normally find on an arrivals shuttle.
- Once people spawn at Departures, they take the arrivals shuttle to the station. The arrivals shuttle is regular, if slightly slow, meaning new arrivals tend to come in waves to the station. When the shuttle leaves the station, it is repaired (read: deleted and spawned anew from template), ensuring no permament damage to the arrivals shuttle, and an end to spawning and instantly dying from NO AIR.
- The shuttle will not leave back to Centcom while people are still on it; after a certain point, it ejects them and/or kills them.
- Observers will continue to arrive near the arrivals dock on that shiny gold memorial plaque.
### Policy Concerns
- It is likely that admins will have to rule that the Departures Area is also covered under no spawn camping/murdering rules, because someone dropping a syndicate bomb there is just rude.

### CODE REVIEW

**Comment by Thunder12345** [comment]
Why are there two seperate pieces and how are they related?

**Comment by coiax** [comment]
Shuttle goes in the middle between the two pieces. They're joined by a corridor on the side.

**Comment by lzimann** [comment]
why though

**Comment by ike709** [comment]
I like it...

**Comment by TechnoAlchemisto** [comment]
kinda cool, reminiscent of lifeweb

**Comment by MMMiracles** [comment]
It's a neat idea, makes a hell of a lot more sense than phasing into existence on the shuttle. As long as the spawn area is a peaceful zone that people aren't allowed to loiter in for long, it could work out well.

**Comment by Durkel** [comment]
Should add a news caster so new arrivals can catch up with what's happening on the station.

**Comment by KorPhaeron** [comment]
This raises some serious lore questions

**Comment by AdamElTablawy** [comment]
such as 'why don't we just use the arrivals shuttle instead of the evac shuttle to escape'

**Comment by Armhulen** [comment]
my immershuns require it and hate it, i'm so confused

**Comment by RandomMarine** [comment]
@MMMiracles 
>As long as the spawn area is a peaceful zone that people aren't allowed to loiter in for long, it could work out well
Some extra leeway should be allowed for new players still getting used to the controls.

**Comment by coiax** [comment]
Provided they're not punching passers by, they can probably loiter as long as they want.

**Comment by Jalleo** [comment]
@KorPhaeron not really

**Comment by Thunder12345** [comment]
@coiax I only see the hallway on one side though?

**Comment by Bawhoppen** [comment]
:-1: As I have said before, there are numerous gameplay functionality as well as thematic problems with this. This will cause all sorts of adminstration problems, as well as possible exploits to get into the rest of z2, not to mention it really doesn't make any sense why they would start at centcomm from a lore perspective.
I recommend that people should just start out in a shuttle flying to the station already, as opposed to this.

**Comment by coiax** [comment]
There are no exploits, it's all surrounded in indestructible walls, the ferry console is unemaggable and will only respect a Centcom ID. From a lore perspective it's about a draw from phasing into existence on the station.

**Comment by Bawhoppen** [comment]
there are potential exploits that are currently unknown, thats what I'm saying
no other time except in shuttle transit (which is essentially safe) do you really get to go to z2

**Comment by coiax** [comment]
z2 is largely uninteresting, large parts of it are boxed off with invulnerable wall, and bug abuse is bannable.

**Comment by ChangelingRain** [comment]
How does the 'limit people from staying on the shuttle' work, exactly?

**Comment by lzimann** [comment]
i really don't see the need for this

**Comment by FlavoredCactus** [comment]
Adding more work on admins for a feature that doesn't really offer more than lore/fluff is probably a bad route.

If this is going to be added i recommend giving users GODMODE without CANPUSH to prevent the clown from camping for sec IDs. as well as making all the turfs indestructible.

Alternatively you can go the goon route and setup teleporter pads for alternative arrival points so you CAN arrive at arrivals but also arrive at dorms/bar/central if it is reported that those areas dont have breathable atmos.

**Comment by GunHog** [comment]
Another consideration would be to withhold human antag announcement until they dock. Since I am given to believe that antag code runs before job assignment (which allows Sec players to retain their antag options), you could hold their names in the antag list (the ticker?) and apply the conversion later. 

But, that is probably too complicated and buggy. We already have code for preventing people attacking others before the game starts, so perhaps it could be extended to being present in the starting areas.

**Comment by coiax** [comment]
@GunHog this is late arrivals only, round start spawn is unaffected.

**Comment by Niknakflak** [comment]
I just think its ugly

**Comment by ChangelingRain** [comment]
What about latejoin/midround-assigned antags, actually? Like double agent loops being created, joining as an antag, etc

**Comment by coiax** [comment]
I haven't touched or modified any midgame antag join code.

Murdering people at Departures is as bad as murdering them on the arrivals shuttle.

**Comment by KorPhaeron** [comment]
We added z checks to stop it grabbing admins anywaay @changelingrain

File: code/game/machinery/status_display.dm [hunk 1]
```diff
@@ -37,6 +37,7 @@
 	var/shuttle_id				// Id used for "generic shuttle timer" mode
 
 	var/friendc = 0      // track if Friend Computer mode
```
Comment (Shadowlight213) [inline code/game/machinery/status_display.dm]: Unrelated to the pr, but wasn't the friend computer button removed?

**Comment by optimumtact** [comment]
So a new players experience is that they're going to spawn in a tiny area, with no visible way to get on the station or explanation for what the hell is going on?

not to mention now as a latespawn you have to sit around for x amount of time (how long does the shuttle take to transit?) before you can even do anything remotely interesting.

How do they interact with the shuttle, do they call it to centcomm or does it just come at a regular time? What kind of timings are we looking at here.


I don’t' see any gameplay benefit that indicates to me that this is worth merging and it's tradeoffs are not worth the RP potential from what I can see.

**Comment by Saegrimr** [comment]
Honestly this seems kinda cool, and solves a bit of that spawncamping on-station during chaos, and is easier to enforce no spawncamping on centcom because if you're attacking/being attacked there at all its not a question of "Well were you actually just arriving or did you run out or run back in or was it bad timing?"

orangEs brings up a valid concern about new players arriving in a tiny room with no way out until x amount of time. That could be fixed by having some kind of spawn text though but they're still gonna have to wait for that shit.

**Comment by optimumtact** [comment]
Hello, respond soonest.

**Comment by Bawhoppen** [comment]
Best solution is to make them already in transit to the station on the shuttle, or the even more simple solution of just making the arrival shuttle turfs respawn

**Comment by optimumtact** [comment]
I know that this is gonna hold the PR up even more but this will really need the assent of the next crop of headmins after they consult their admins, as they're gonna bear the brunt of managing and administering the area.

**Comment by Niknakflak** [comment]
It's still a very ugly map too

**Comment by Shadowlight213** [comment]
Just make it a config option.

**Comment by Armhulen** [comment]
thats what i said about memmememem but where is that now?

**Comment by optimumtact** [comment]
Stale, ping me to reopen or make a new pr when ready

**Pull request closed without merging**